DIPs/DIP1009.md

```diff
@@ -42,38 +44,51 @@ body
+Experience demonstrates, as in the case of `=>` lambdas, that a small change in syntax can dramatically affect the usage of a language feature. By adding a new syntax for contracts, a similar increase in their usage can be expected.
 
 Furthermore, because the existing `invariant` grammar is so similar to `in` and `out` contract grammar, any improvements to `in` and `out` contracts can be transferred easily to invariants as well. Consistency across language constructs is desirable.
 
 ## Description
 
-Allow `in`, `out`, and `invariant` contracts to be written as parenthesized expressions. Allow multiple expressions of this kind. Create a new grammatical construct called ContractParameters, which copies and replaces the parameters to `assert` expressions. Use the new construct for `in`, `out`, `assert`, and `invariant` expressions alike. `in` and `out` expressions must come at the end of the function declarator suffix, and before the regular contracts, if any. `invariant` expressions require a semicolon terminator. (Body-less interface functions which use only the new syntax also require a semicolon terminator.) The keyword `do` or `body` is not required after the new `in` and `out` expressions.  The choice of syntax for `out` expressions requires a separate analysis (see the Analysis section below). For the sake of the following example, the `out` expression syntax requires a semicolon separating an optional return identifier from its subsequent ContractParameters:
+Take the grammar for existing `assert` parameters, and call it ContractParameters. Copy this grammar and use it in all the new syntaxes.
+
+Introduce two new `in` and `out` contract syntaxes, called InExpression and OutExpression.
+
+An InExpression is `in`, followed by parentheses containing ContractParameters, exactly mimicking an `assert` call.
+
+The recommended OutExpression syntax is `out`, followed by parentheses containing an optional return identifier, a manditory semicolon, and ContractParameters (see Analysis below).
 
+For example:
 ```d
 int fun(ref int a, int b)
    in(a > 0)
    in(b >= 0, "b cannot be negative!")
    out(r; r > 0, "return must be positive")
-   out(; a !=0)
+   out(; a != 0)
 {
    // function body
 }
-
-class C {
-   int data;
-   invariant(data != 0, "data cannot be 0");
-}
-
+```
+InExpressions and OutExpressions are considered the last part of the function signature. Multiple such expressions are permitted. The existing grammar for the function body, which includes ordinary bracketed `in` and `out` contracts, is unchanged. Therefore no `do` or `body` keyword is required after a new InExpression or OutExpression, unless it is also followed by a bracketed `in` or `out` contract. In the case of body-less virtual interface functions, a semicolon is required if there is no bracketed contract:
```

Comment:
"InExpressions and OutExpressions are considered the last part of the function signature. Multiple such expressions are permitted. The existing grammar for the function body, which includes ordinary bracketed `in` and `out` contracts, is unchanged. Therefore no `do` or `body` keyword is required after a new InExpression or OutExpression, unless it is also followed by a bracketed `in` or `out` contract."

With my change above, this could be rewritten to a single sentence explaining that bracketed contracts require a the `do` keyword, followed by a snippet that shows it.